### PR TITLE
Update library-go for support out-of-tree Azure provider

### PR DESCRIPTION
**Pull request by lobziik**
Updated library-go dependency with changes from https://github.com/openshift/library-go/pull/1077. Needed for proceeding with Azure/AzureStackHub out of tree providers related work.

### CODE REVIEW

**Review by Fedosin — APPROVED** [review]
/lgtm

File: pkg/controller/template/render.go [hunk 1]
```diff
@@ -379,7 +379,7 @@ func cloudConfigFlag(cfg RenderConfig) interface{} {
 		}
 	}
 
-	external, err := cloudprovider.IsCloudProviderExternal(cfg.Infra.Status.PlatformStatus.Type, cfg.FeatureGate)
+	external, err := cloudprovider.IsCloudProviderExternal(cfg.Infra.Status.PlatformStatus, cfg.FeatureGate)
```
Comment (kikisdeliveryservice) [inline pkg/controller/template/render.go:382]: For my own info, can you explain this change?

**Review by kikisdeliveryservice — COMMENTED** [review]
1 question :)

File: pkg/controller/template/render.go (hunk 1 above)
Comment (JoelSpeed) [reply pkg/controller/template/render.go:382]: We had to change the signature of the upstream method to include more information from the Infrastructure resource to allow the method to determine whether it is on a special cloud, in this case Azure stack hub (this also applies for things like gov clouds etc), where the platform shows up as Azure, but the designation that this is Azure Stack Hub is only available by looking at the Azure specific fields on the `PlatformStatus`. We don't expect this to change again in the future as we now have access to all of the provider specific statuses to make the decision needed in the future

Comment (kikisdeliveryservice) [reply pkg/controller/template/render.go:382]: thanks for the details @JoelSpeed makes sense to me.

Comment (kikisdeliveryservice) [reply pkg/controller/template/render.go:382]: as yes it's right here :smile: 
https://github.com/openshift/library-go/pull/1077/files#diff-478af36e9fb994fc80d37b7d2f6ae207c67d8c43b94f98f6ae3e420808958ba9

**Review by kikisdeliveryservice — APPROVED** [review]
Changes make sense and are in line with library-go changes for Azure

**Pull request merged**